tests/cpp/test_pointwise.cpp

```diff
@@ -478,12 +478,106 @@ TEST_F(PointwiseTest, VIssue1567ectorizationFactorAnalysisCase3) {
   fe.compileFusion(fusion, aten_inputs, lparams);
   auto cg_outputs = fe.runFusion(aten_inputs, lparams);
 
-  EXPECT_EQ(params->vectorize, true);
+  EXPECT_TRUE(params->vectorize);
   EXPECT_EQ(params->unroll_factor, 2);
   EXPECT_TRUE(hasVectorizationCache(tv0));
   EXPECT_TRUE(hasVectorizationCache(tv1));
 
   testValidate(fusion, cg_outputs, aten_inputs, __LINE__, __FILE__);
 }
 
+// params: use 1D scheduler, sharded dimenson
+class ShardedPointwiseTest
+    : public NVFuserTest,
+      public testing::WithParamInterface<std::tuple<bool, int>> {};
+
+TEST_P(ShardedPointwiseTest, DID_Compatible) {
+  auto [use_1D_scheduler, sharded_dim] = GetParam();
+  auto fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  TensorView* tv0 = makeContigTensor(4);
+  TensorView* tv1 = makeContigTensor(2);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = add(tv0, tv0);
+  auto tv3 = broadcast(tv1, {true, true, false, false});
+  auto tv4 = add(tv2, tv3);
+  fusion->addOutput(tv4);
+
+  DeviceMesh mesh = DeviceMesh::createForNumDevices(4);
+  for (TensorView* tv : {tv0, tv2, tv3, tv4}) {
+    tv->setDeviceMesh(mesh);
+    tv->axis(sharded_dim)->parallelize(ParallelType::DIDx);
+  }
+  tv1->setDeviceMesh(mesh);
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
+  fec.profile(true);
+
+  // Trigger the 1D scheduler by using small input size.
+  std::vector<int64_t> input_size(3);
+  if (use_1D_scheduler) {
+    input_size = {16, 8, 24};
+  } else {
+    input_size = {1024, 32, 64};
+  }
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn(input_size, options).unsqueeze(sharded_dim);
+  at::Tensor t1 = at::randn({input_size[1], input_size[2]}, options);
+  std::vector<c10::IValue> aten_inputs = {t0, t1};
+
+  auto params = getPointwiseHeuristics(fusion, aten_inputs);
+  auto lparams = schedulePointwise(fusion, aten_inputs);
+  FusionExecutor fe;
+  fe.compileFusion(fusion, aten_inputs, lparams);
+  auto cg_outputs = fe.runFusion(aten_inputs, lparams);
+
+  EXPECT_EQ(use_1D_scheduler, params->break_point == 0);
+  testValidate(fusion, cg_outputs, aten_inputs, __LINE__, __FILE__);
+
+  // Create a single device fusion and verify the same scheduling parameters are
+  // used for given the same single device fusion.
+  auto unsharded_fusion_ptr = std::make_unique<Fusion>();
+  auto unsharded_fusion = unsharded_fusion_ptr.get();
+  FusionGuard unsharded_fg(unsharded_fusion);
+
+  TensorView* utv0 = makeContigTensor(3);
```

Comment:
Any particular reason to use 3? I think [16,1,8,24] as a reference fusion would work too, and it may simplify the test by only requiring to clone the fusion not building a new fusion from scratch.

Reply:
I used 3 to keep the number of non-device dim axes the same. This is important just for testing when we check the break_points are the same. 
I agree that right now the test is a bit clunky because we are creating two slightly different fusions (also easy to make a simple mistake when editing), but when we allow DID with split, we can use a single fusion with input size [16, 8, 24] and have one with DID scheduling and one without.

Reply:
If we were to use `[16,1,8,24]`, `params->break_point` and `unsharded_params->break_point` would be off by one? That would make sense to me. 

Nonetheless, you can still put the creation of the fusion into a helper function to reduce duplication. E.g. 

```
auto create_fusion = [](const std::vector<int64_t>& input_shape) -> std::unique_ptr<Fusion> {
  ...
};
```